src/main/java/nl/hsac/fitnesse/fixture/util/HttpResponse.java

```diff
@@ -31,12 +32,35 @@ public void validResponse() {
         if (statusCode < 100) {
             throw new RuntimeException("Status code is less than 100: " + statusCode);
         }
+        if (statusCode >= 400 && statusCode <= 499) {
+            throw new RuntimeException("Server reported client error: " + statusCode);
+        }
         if (statusCode >= 500 && statusCode <= 599) {
             throw new RuntimeException("Server error returned: " + statusCode);
         }
-        if (statusCode >= 400 && statusCode <= 499) {
-            throw new RuntimeException("Server reported client error: " + statusCode);
+    }
+
+    public boolean responseIsValid(boolean throwExceptionOnHttpRequestFailure) {
```

Comment:
why not just use the implementation of the existing method and catch the RuntimeException and convert that to a SlimFixtureException if boolean is true

Reply:
I did not want to use (or touch) this method, because throwing a RuntimeException without specifing what kind of runtime exception seems like a pretty big smell to me. And catching *any* runtime exception and converting it to something specific does not make it better if you ask me.

Reply:
You are right about catching a general RuntimeException. I never used the 'slim' exception here because this code is also using from Fit fixtures, and Fit has no special handling for SlimFixtureException.

But we can probably to it the other way around. Define a custom subclass of RuntimeException (e.g. 'NonValidResponseReceivedException'). And have this thrown by the request methods. In the slim fixtures we can catch this and either convert it to 'false' or to a SlimFixtureException.
What are your thoughts on that?